core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
The problem with having these checks in validate() is that when request comes through rest layer, null values here are okay here. They are just not okay when rendering the request as XContent in the builder.